server/plugin.go

```diff
@@ -204,6 +296,86 @@ func (p *Plugin) updateJwtUserInfo(jwtToken string, user *model.User) (string, e
 	return signClaims(secret, claims)
 }
 
+func (p *Plugin) setJWTClaims(user *model.User, userType string) (string, error) {
+	// User did not specify a jwt, generate a new jwt for user
+	const ExpTimeDelaySec = 7200
+	permission := "true"
+	if userType != typeUser {
+		permission = "false"
+	}
+
+	claims := JaaSClaims{}
+	claims.Issuer = jaasIssuerClaim
+	claims.Audience = jaasAudienceClaim
+	claims.Subject = p.getConfiguration().JaaSAppID
+	claims.Room = "*"
+	claims.Exp = time.Now().Unix() + ExpTimeDelaySec
+	claims.Nbf = time.Now().Unix()
+	claims.Context.Features.LiveStreaming = permission
+	claims.Context.Features.Recording = permission
+	claims.Context.Features.OutboundCall = permission
+	claims.Context.Features.Transcription = permission
+	claims.Context.User.Avatar = ""
+	claims.Context.User.Email = ""
+	if userType == typeUser {
+		claims.Context.User.Email = user.Email
+	}
+	claims.Context.User.ID = user.Id
+	claims.Context.User.Moderator = permission
+	claims.Context.User.Name = user.GetFullName()
+
+	jwtToken, err := signClaimsJaaS(p.getConfiguration().JaaSApiKey, p.getConfiguration().JaaSPrivateKey, &claims)
+	if err != nil {
+		mlog.Error(fmt.Sprintf("Error generating JaaS token for %s", userType), mlog.Err(err))
+		return "", errors.Wrap(err, fmt.Sprintf("failed creating new JaaS token for %s %s", userType, user.Id))
+	}
+
+	return jwtToken, nil
+}
+
+func (p *Plugin) generateJaaSJwtForUser(user *model.User) (string, error) {
+	return p.setJWTClaims(user, typeUser)
+}
+
+func (p *Plugin) generateJaaSJwtForGuest(userid string) (string, error) {
+	user := &model.User{
+		Id: userid,
+	}
+
+	return p.setJWTClaims(user, typeGuest)
+}
+
+func (p *Plugin) getJaaSSettings(jwtToken string, path string, user *model.User) (*JaaSSettings, error) {
+	if user != nil {
+		claims, err := verifyJwtJaaS(jwtToken)
+		if err != nil {
+			jwtToken, err = p.generateJaaSJwtForUser(user)
```

Comment:
Is the correct thing to do to generate a new JWT for any error that occurs in `verifyJwtJaaS`?

Reply:
@raghavaggarwal2308 What are you thoughts on my question above?

Reply:
@mickmister ```verifyJwtJaaS``` is a misleading name for this function that's why I have changed the name here, we are trying to get the claims for the token and if we are unable to do so then we are generating a new token for that mattermost user.

Reply:
> we are trying to get the claims for the token and if we are unable to do so then we are generating a new token for that mattermost user.

@Kshitij-Katiyar What if this is a "bad request"? In what cases are we unable to get the claims, causing this error?

Reply:
@mickmister There could be many cases where we can get this error:-
1) When the token is in the wrong format. For example, jwt token should start with ```eyJ```, or we aren't able to base64 decode the token.
2) We can get an error while unmarshalling the claims of the jwt token.